src/mmw/js/src/modeling/models.js

```diff
@@ -833,6 +854,27 @@ var ScenarioModel = Backbone.Model.extend({
         return taskModel.start(taskHelper);
     },
 
+    // If the model has never completed, we wont be able to run
+    // it with modifications
+    // If there's already a `fetchResult` in progress, wait for it
+    // to complete before trying to run the model with the modifications
+    // If there's none in progress, call the initial fetchResultsIfNeeded
+    // Otherwise just fetch results as usual
+    fetchResultsAfterInitial: function() {
```

Comment:
Hm, yeah that makes a lot of sense! I thought there had been an issue involving two `fetchResults` promises chaining to the `fetchResultsPromise` (one here and one from the initial call from the view), and the original `modification_censuses` error returning. I can't recreate it. It was probably wrapped up in some other silly thing I was doing earlier :)